kong/tracing/instrumentation.lua

```diff
@@ -0,0 +1,353 @@
+local pdk_tracer = require "kong.pdk.tracing".new()
+local utils = require "kong.tools.utils"
+local tablepool = require "tablepool"
+local tablex = require "pl.tablex"
+local base = require "resty.core.base"
+local cjson = require "cjson"
+local ngx_re = require "ngx.re"
+
+local ngx = ngx
+local var = ngx.var
+local pack = utils.pack
+local unpack = utils.unpack
+local insert = table.insert
+local new_tab = base.new_tab
+local time_ns = utils.time_ns
+local tablepool_release = tablepool.release
+local get_method = ngx.req.get_method
+local ngx_log = ngx.log
+local ngx_DEBUG = ngx.DEBUG
+local concat = table.concat
+local cjson_encode = cjson.encode
+local _log_prefix = "[tracing] "
+local split = ngx_re.split
+
+local _M = {}
+local tracer = pdk_tracer
+local NOOP = function() end
+local available_types = {}
+
+local POOL_SPAN_STORAGE = "KONG_SPAN_STORAGE"
+
+-- Record DB query
+function _M.db_query(connector)
+  local f = connector.query
```

Comment:
Naming could be better.
And what if we save the original function, like
```lua
connector.query_origin = connector.query
```
So we can escape from tracing when needed?

For example:

`http_export_request` calls `request_uri` which is hooked here.

Reply:
I think there is no way back to the original function when we have patched :(